### ISSUE TITLE
phone app inaccurate tracking

### ISSUE THREAD

**Issue by rybalko** [e-mission/e-mission-docs#694]
I tested phone app on android smartphone today
it successfully automatically caught start trip and stop trip points,
but tracking in general was not accurate, see Screenshot
![Screenshot_20211209-164144_emission](https://user-images.githubusercontent.com/357010/145443048-22f34aed-ffcc-411e-a39e-de481b56c8cc.jpg)

I marked with blue square and circle those wrong parts of the tracking,
and 6 km in 22 minutes is not right.

**Comment by shankari** [comment]
@rybalko we get the raw points directly from the phone OS (android in this case). You can look at the logs or the `background/location` entries in the mongodb to see what they are. Unfortunately, this raw data can frequently be noisy - we can't fix that without modifying the OS.

We do try to clean up obvious zig-zags - see sections 5.2 and 5.3 (in particular, 5.3.3) for a description of the data characteristics and our attempts to clean it up. Section 7.2.8 has some additional examples and a more statistical summary.

To summarize, we are never going to be able to remove all noise. There will be some trips that have incorrect values. We do our best, but the example above is particularly hard to address because the zigzag appears to be small and short. If you would like to tackle this problem and contribute a better algorithm, it would be welcome!

**Comment by shankari** [comment]
@rybalko any updates on this, or can we close this issue?

**Comment by rybalko** [comment]
I tested appp one more time, but still have strange hAcc fluctuations up to hAcc=1400
but mostly points have hAcc between 10 and 60, see log
````
17163,1639123368.37,2021-12-10T10:02:48.370000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.947993,33.444773 hAcc=10 et=+40d21h49m53s379ms alt=106.55168995844217 vel=2.24 bear=241.0 vAcc=3 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17196,1639123400.87,2021-12-10T10:03:20.870000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.947623,33.444283 hAcc=10 et=+40d21h50m25s880ms alt=106.85104004819095 vel=2.1599998 bear=232.0 vAcc=3 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17229,1639123433.708,2021-12-10T10:03:53.708000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.947623,33.444283 hAcc=56 et=+40d21h50m56s325ms alt=113.49322867674545 vAcc=3 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17262,1639123466.682,2021-12-10T10:04:26.682000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.945675,33.441876 hAcc=1400 et=+40d21h51m29s311ms alt=113.41918158330952 vAcc=3 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}]] from intent"
17285,1639123498.889,2021-12-10T10:04:58.889000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.946561,33.442842 hAcc=10 et=+40d21h52m3s878ms alt=99.20047319327516 vel=1.68 bear=0.0 vAcc=94 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17315,1639123531.579,2021-12-10T10:05:31.579000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.946561,33.442842 hAcc=55 et=+40d21h52m34s339ms alt=110.54696988035406 vAcc=3 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17348,1639123564.293,2021-12-10T10:06:04.293000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.945619,33.441696 hAcc=12 et=+40d21h53m9s305ms alt=99.2302069003652 vel=1.12 bear=0.0 vAcc=120 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
17381,1639123597.118,2021-12-10T10:06:37.118000+02:00,"LocationChangeIntentService : Read locations [Location[fused 47.945619,33.441696 hAcc=60 et=+40d21h53m41s535ms alt=101.54544263130703 vAcc=3 sAcc=20 bAcc=10 {Bundle[mParcelledData.dataSize=52]}]] from intent"
````
and because this big inaccuracy a track on the map has those strange zigzags
There is GPS Update interval seems like 25 seconds

I tried couple other gps tracking apps
GPS Logger Update interval:  1 sec
Urban Biker Update interval:  2 sec

GPS Logger interval cannot be set more then 3 sec

I don't have access to those apps raw data, but result GPX on the map looks very accurate

**Comment by shankari** [comment]
@rybalko

>  There is GPS Update interval seems like 25 seconds

Yes, the GPS update interval is intentionally set to 30 secs to reduce the power drain. Please see the power evaluation in section 7.2 of my thesis. If you want to experiment with different frequencies, you can change the settings in the Profile -> Developer Zone section.

From Figure 7.14, I do see that, after excluding outliers, the error on android is higher with medium frequency (30secs) than with high frequency (1sec). Would be good to see what your results are.

> and because this big inaccuracy a track on the map has those strange zigzags

I am not convinced that it is. Note that we only store `filtered_location` objects for points whose accuracy is below the threshold.
storing: https://github.com/e-mission/e-mission-data-collection/blob/e152c945fd29ae7ba89e4c9df6c5ab79346391af/src/android/location/LocationChangeIntentService.java#L168
accuracy constant: https://github.com/e-mission/e-mission-data-collection/blob/e152c945fd29ae7ba89e4c9df6c5ab79346391af/src/android/wrapper/LocationTrackingConfig.java#L18

And we use the filtered_locations in all further pipeline steps. The server mongodb actually has both the locations (`background/location`) and the filtered locations (`background/filtered_location`) separately if you want to confirm that the filtering works 😄 

The zigzags (if any) are from points which are marked as accurate but are actually inaccurate.
Again, on the server, you can print out the filtered location points within the trip start and end to see the points that the analysis is using in greater detail.

The "Can you do better" section of https://github.com/e-mission/e-mission-server/blob/master/Timeseries_Sample.ipynb has some sample code.

Happy to look at the results of that investigation for the trip above and provide additional insights.

**Comment by shankari** [comment]
closing this since it hasn't been updated for a while.
